File: data/src/main/java/com/roman/kubik/spivanyklicejista/data/history/HistoryDao.java
package com.roman.kubik.spivanyklicejista.data.history;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import java.util.List;

import io.reactivex.Single;


@Dao
public interface HistoryDao {

    @Query("SELECT * FROM history")
    Single<List<HistoryEntity>> getHistory();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void addToHistory(HistoryEntity historyEntity);
}

File: songsRoom/src/main/java/com/roman/kubik/songer/room/history/HistoryDao.kt
package com.roman.kubik.songer.room.history

import androidx.room.Dao
import androidx.room.Query
import com.roman.kubik.songer.room.song.SongEntity


@Dao
interface HistoryDao {

    @Query("SELECT * FROM history INNER JOIN song ON song.id = history.song_id ORDER BY timestamp DESC")
    suspend fun getHistory(): List<SongEntity>

}